test/utils/ruleset-test-harness.ts

```diff
@@ -0,0 +1,151 @@
+/* eslint-disable class-methods-use-this */
+/* eslint-disable no-underscore-dangle */
+import 'mocha';
+import {
+  Browser, Page, chromium, firefox, webkit,
+} from 'playwright';
+
+import {
+  expectFS, ExpectObserver, setupGlobals, expectGlobal,
+} from './mocha';
+
+export interface RulesetTestHarness {
+  setUp: (rules: any, dataLayer: any) => Promise<void>;
```

Comment:
Rules should really be of type `DataLayerRule[]`. The initialization is a little weird since we're importing a JS file with the ruleset being assigned to the global `window` object -- then pulling the rules out of the `window` object typed as `any`. To increase clarity, though, I've changed `rules` within the ruleset test utils to be typed as `DataLayerRule[]`.